CodeEdit/Features/Settings/Pages/ThemeSettings/Models/ThemeModel+CRUD.swift

```diff
@@ -0,0 +1,292 @@
+            var newFileName = fileName
+
+            // Check if the file already exists
+            var iterator = 1
+
+            let isBundled = url.absoluteString.hasPrefix(bundledThemesURL?.absoluteString ?? "")
+            let isImporting =
+                !url.absoluteString.hasPrefix(bundledThemesURL?.absoluteString ?? "")
+                && !url.absoluteString.hasPrefix(themesURL.absoluteString)
+
+            if isBundled {
+                newFileName = "\(fileName) \(iterator)"
+                destinationFileURL = self.themesURL
+                    .appendingPathComponent(newFileName)
+                    .appendingPathExtension(fileExtension)
+            }
+
+            while FileManager.default.fileExists(atPath: destinationFileURL.path) {
+                fileName = destinationFileURL.deletingPathExtension().lastPathComponent
+
+                // Remove any existing iterator
+                if let range = fileName.range(of: " \\d+$", options: .regularExpression) {
+                    fileName = String(fileName[..<range.lowerBound])
+                }
+
+                // Generate a new filename with an iterator
+                newFileName = "\(fileName) \(iterator)"
+                destinationFileURL = self.themesURL
+                    .appendingPathComponent(newFileName)
+                    .appendingPathExtension(fileExtension)
+
+                iterator += 1
+            }
+
+            // Copy the file from selected URL to the destination
+            try FileManager.default.copyItem(at: url, to: destinationFileURL)
+
+            try self.loadThemes()
+
+            if var index = self.themes.firstIndex(where: { $0.fileURL == destinationFileURL }) {
+                self.themes[index].displayName = newFileName
+                self.themes[index].name = newFileName.lowercased().replacingOccurrences(of: " ", with: "-")
+                if isImporting != true {
+                    self.themes[index].author = NSFullUserName()
+                    self.save(self.themes[index])
+                }
+                self.selectedTheme = self.themes[index]
+                self.detailsTheme = self.themes[index]
+            }
+        } catch {
+            print("Error adding theme: \(error.localizedDescription)")
+        }
+    }
+
+    func rename(to newName: String, theme: Theme) {
+        do {
+            guard let oldURL = theme.fileURL else {
+                throw NSError(
+                    domain: "ThemeModel",
+                    code: 1,
+                    userInfo: [NSLocalizedDescriptionKey: "Theme file URL not found"]
+                )
+            }
+
+            var iterator = 1
+            var finalName = newName
+            var finalURL = themesURL.appendingPathComponent(finalName).appendingPathExtension("cetheme")
+
+            // Check for existing display names in themes
+            while themes.contains(where: { theme != $0 && $0.displayName == finalName }) {
```

Comment:
Lets make this into a generic, reusable function since it's being used multiple times.

Reply:
They are a actually different, one is checking the filename, the other is checking the names across all the themes.